ovos_PHAL_plugin_oauth/__init__.py

```diff
@@ -182,6 +188,47 @@ def handle_oauth_register(self, message):
             # eg. spotify
             self.oauth_skills[skill_id]["needs_creds"] = True
 
+    def handle_refresh_token(self, message):
+        """Refresh oauth token.
+
+        See:
+        https://www.oauth.com/oauth2-servers/making-authenticated-requests/refreshing-an-access-token/
+
+        for details on the procedure.
+        """
+        oauth_id = f"{message.data['skill_id']}_{message.data['app_id']}"
+        # Load all needed data for refresh
+        with OAuthApplicationDatabase() as db:
+            app_data = db[oauth_id]
+        with OAuthTokenDatabase() as db:
+            token_data = db[oauth_id]
+
+        if 'refresh_token' not in token_data:
+            LOG.warning("Token data doesn't contain a refresh token and cannot "
+                        "be refreshed.")
+            return
+
+        refresh_token = token_data['refresh_token']
+        refresh_endpoint = app_data["refresh_endpoint"]
```

Comment:
Maybe changing this to 

`refresh_endpoint = app_data.get("refresh_endpoint") or app_data["token_endpoint"]`

as at least google with the newer API versions uses the token endpoint for both.

Reply:
Yes, that might be a bit more failsafe. For spotify I set token endpoint as refresh endpoint during initial setup so I assumed it would be left to the user/dev to set it up properly.

Reply:
Reading up a bit on oauth2 again. The refresh endpoint was copied from spotipy's implementation but according to https://auth0.com/docs/authenticate/protocols/oauth it should always be the token endpoint. So I think the refresh_endpoint should be removed altogether...